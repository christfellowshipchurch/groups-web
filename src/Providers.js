import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
