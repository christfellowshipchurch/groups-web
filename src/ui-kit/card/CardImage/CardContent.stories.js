import React from 'react';

import CardImage from '.';

const Template = (args) => <CardImage {...args} />;

const Story = {
  component: CardImage,
  title: 'ui-kit/card/CardImage',
};

const example = Template.bind({});
example.args = {
  src: 'https://picsum.photos/840/472',
};

const isLoading = Template.bind({});
isLoading.args = {
  isLoading: true,
};

export { Story as default, example, isLoading };
