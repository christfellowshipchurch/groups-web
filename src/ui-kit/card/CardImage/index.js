import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Image from '../../Image';
import { getIsLoading } from '../../isLoading';
import { Media } from '../../Placeholder';
import { withOverrides } from '../../theme';

const Placeholder = styled(Media)`
  border-radius: ${({ theme }) =>
    `${theme.sizing.baseBorderRadius} ${theme.sizing.baseBorderRadius} 0 0`};
`;

const StyledImage = styled(Image).attrs(
  withOverrides('ui-kit.card.CardImage.StyledImage')
)``;

const CardImage = getIsLoading(({ isLoading, ...props }) => (
  <Placeholder isLoading={isLoading} showLoadingAnimation>
    <StyledImage {...props} />
  </Placeholder>
));

CardImage.propTypes = {
  isLoading: PropTypes.bool,
};

CardImage.displayName = 'ui-kit.card.CardImage';

export default CardImage;
