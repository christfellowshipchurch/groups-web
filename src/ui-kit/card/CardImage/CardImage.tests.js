import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import CardImage from '.';

describe('the CardImage component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <CardImage src={'https://picsum.photos/300/200'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state (isLoading)', () => {
    const tree = renderer.create(
      <Providers>
        <CardImage src={'https://picsum.photos/300/200'} isLoading />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
