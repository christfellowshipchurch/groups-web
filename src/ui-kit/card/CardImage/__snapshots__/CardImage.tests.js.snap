// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CardImage component should render 1`] = `
.c0 {
  width: 100%;
  height: auto;
}

<img
  className="c0 sc-1shebyi-1 kaSgGP"
  src="https://picsum.photos/300/200"
/>
`;

exports[`the CardImage component should render a loading state (isLoading) 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding: 100% 0 0 0;
}

.c1 {
  border-radius: 5px 5px 0 0;
}

<mark
  className="c0 j5u57d-1 c1 custom-placeholder"
  size="100%"
/>
`;
